dev/update_mlflow_versions.py

```diff
@@ -113,12 +121,16 @@ def pre_release(new_version: str):
 )
 def post_release(new_version: str):
     current_version = Version(get_current_py_version())
-    assert (
-        current_version.is_devrelease
-    ), f"{current_version} is not a dev version. This script must be run on master branch."
```

Comment:
`get_current_py_version` currently strips a dev suffix so the assertion never succeeds. This PR fixes it.